src/aunit/AssertMacros.h

```diff
@@ -38,35 +38,35 @@ SOFTWARE.
 
 /** Assert that arg1 is equal to arg2. */
 #define assertEqual(arg1,arg2) \
-    assertOpInternal(arg1,aunit::internal::compareEqual,"==",arg2)
+    assertOpInternal((arg1),aunit::internal::compareEqual,"==",(arg2))
```

Comment:
Just so that I understand, the bug occurs *only* when using a comma-operator inside one of the arguments, right? Because every other C++ operator has a higher precedence than the comma-operator, so the current code does not normally trigger a bug.